### PR TITLE
docs: Add missing proxy and fastcgi options

### PR DESCRIPTION
**Pull request by francislavoie**
Some of these are v2.3 things, others were just forgotten (`resolve_root_symlink` for example)

### CODE REVIEW

File: src/docs/markdown/caddyfile/directives/reverse_proxy.md [hunk 1]
```diff
@@ -133,13 +133,18 @@ Caddy's proxy **transport** is pluggable:
...
-	read_buffer  <size>
-	write_buffer <size>
-	dial_timeout <duration>
+	read_buffer             <size>
+	write_buffer            <size>
+    max_response_header     <size>
+	dial_timeout            <duration>
+    dial_fallback_delay     <duration>
+    response_header_timeout <duration>
+    expect_continue_timeout <duration>
```
Comment (mholt) [inline src/docs/markdown/caddyfile/directives/reverse_proxy.md]:
```suggestion
	read_buffer             <size>
	write_buffer            <size>
	max_response_header     <size>
	dial_timeout            <duration>
	dial_fallback_delay     <duration>
	response_header_timeout <duration>
	expect_continue_timeout <duration>
```

Those naughty spaces again

**Review by mholt — CHANGES_REQUESTED** [review]
Thank you for this! But get your tab key fixed :smile:

File: src/docs/markdown/caddyfile/directives/reverse_proxy.md (hunk 1 above)
Comment (francislavoie) [reply src/docs/markdown/caddyfile/directives/reverse_proxy.md]: Dangit sorry, we should add an `.editorconfig` in this repo tbh

File: src/docs/markdown/caddyfile/directives/reverse_proxy.md [hunk 2]
```diff
@@ -179,13 +190,19 @@ transport fastcgi {
 	split <at>
 	env   <key> <value>
     resolve_root_symlink
+    dial_timeout  <duration>
+	read_timeout  <duration>
+	write_timeout <duration>
```
Comment (mholt) [inline src/docs/markdown/caddyfile/directives/reverse_proxy.md]:
```suggestion
	dial_timeout  <duration>
	read_timeout  <duration>
	write_timeout <duration>
```

File: src/docs/markdown/caddyfile/directives/reverse_proxy.md [hunk 3]
```diff
@@ -179,13 +190,19 @@ transport fastcgi {
 	split <at>
 	env   <key> <value>
     resolve_root_symlink
+	dial_timeout  <duration>
+	read_timeout  <duration>
+	write_timeout <duration>
```
Comment (mholt) [inline src/docs/markdown/caddyfile/directives/reverse_proxy.md]:
```suggestion
	resolve_root_symlink
	dial_timeout  <duration>
	read_timeout  <duration>
	write_timeout <duration>
```

I don't know how this keeps happening

**Review by mholt — APPROVED** [review]

**Pull request merged**